iModelCore/GeomLibs/PublicAPI/GeomSerialization/GeomLibsFlatBufferApi.h

```diff
@@ -22,25 +22,55 @@ BEGIN_BENTLEY_GEOMETRY_NAMESPACE
 class BentleyGeometryFlatBuffer
 {
 private:
-BentleyGeometryFlatBuffer (){} // no intances.
+BentleyGeometryFlatBuffer (){} // no instances.
+// template <class PtrName, class ReaderMethodName>
+// PtrName BytesToXXXSafe(Byte const *buffer, size_t const bufferSize, bool applyValidation);
+
 public:
 
 // test if the bytes have the signature for BentleyGeomFlatBuffer.
-static BGFBIMPEXP bool IsFlatBufferFormat (bvector <Byte> &buffer);
-static BGFBIMPEXP bool IsFlatBufferFormat (Byte const *buffer);
+static BGFBIMPEXP bool IsFlatBufferFormat(bvector <Byte> &buffer);
+static BGFBIMPEXP bool IsFlatBufferFormat(Byte const *buffer);
+
+static BGFBIMPEXP IGeometryPtr BytesToGeometry(bvector <Byte> &buffer, bool applyValidation = true);
+//! @deprecated in 4.x. Use `BytesToGeometrySafe`.

```

Comment:
Why deprecate? imodel-native APIs are not subject to public API support policies.

Reply:
(my bad; I should have mentioned the reason in the commint message; will add)

The reason for deprecation is those functions are not safe and crash at run time if they receive bad binary files as input. We had change the API argument list so had to deprecate them. We (Dave and I) are working with Mike Belousov to reveal and fix those kind of run time crashes.

Reply:
imodel-native APIs are not public. You are free to delete them, or change their implementations and/or argument lists.

Reply:
For imodel02 I would not expect so. Connectors would update to new version of imodel02 and adjust their code accordingly if needed. right @jffmarker?

Can't comment on PP.

Reply:
This is good to know. Last year we stripped a bunch of stuff from imodel-native geomlibs in order to make it more palatable for open sourcing, so I was operating under some assumptions about its API evolution. We have dozens of deprecations in native geomlibs over the years, but unfortunately no process to remove the deprecated methods. So what I'm reading here is that we could sweep away the deprecations in imodel-native and imode02. PPBase geomlibs is more restrictive: I've been told not to change signatures in the PublicAPI there, and I'm not aware of a process to clean up deprecations there.

Reply:
@pmconne @saeeedtorabi  Correct, for imodel02 consumers, there's no strict API lifecycle. If you make a breaking change, you will be expected to help consumers port or refactor when they get your changes from biweekly builds, but you are generally free to make changes, especially for security concerns.

Reply:
> we could sweep away the deprecations in imodel-native and imode02

Yep.

Reply:
Removed deprecation and applied API chanegs to the existing API. Please "resolve conversation" if you are satisfied with the change.

Reply:
@pmconne